src/components/Microsoft.DotNet.UpgradeAssistant.Extensions/ExtensionManager.cs

```diff
@@ -26,7 +26,14 @@ internal sealed class ExtensionManager : IDisposable, IEnumerable<ExtensionInsta
                 {
                     if (LoadExtension(path) is ExtensionInstance extension)
                     {
-                        list.Add(extension);
+                        if (extension.MinUpgradeAssistantVersion is Version minVersion && minVersion < options.Value.CurrentVersion)
+                        {
+                            logger.LogWarning("Could not load extension from {Path}", path);
```

Comment:
Would be good to call out in the warning that it couldn't be loaded because UA's version was too low and what the min version is.